src/components/pricing/lifetime/upcoming-sale.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react'
+import {useViewer} from '@/context/viewer-context'
+import emailIsValid from '@/utils/email-is-valid'
+import PoweredByStripe from '@/components/pricing/powered-by-stripe'
+import useCio from '@/hooks/use-cio'
+import {INTERESTED_IN_LIFETIME_SINCE} from '@/utils/cio/subscriber-attributes'
+import {requestSignInEmail} from '@/utils/request-signin-email'
+import {cx} from 'class-variance-authority'
+import {CheckCircleIcon} from '@heroicons/react/solid'
+import {PlanTitle} from './active-sale'
+
+const UpcomingSale = () => {
+  const {viewer} = useViewer()
+  const {subscriber, cioIdentify} = useCio()
+
+  const [email, setEmail] = React.useState<string>('')
+  const [signedUp, setSignedUp] = React.useState<boolean>(false)
+
+  async function handleClick(submittedEmail?: string) {
+    setSignedUp(true)
+
+    // there will only be a submittedEmail if there is no viewer.
+    // subscriber can still exist because it looks at cookies in the browser but would be confusing for the user if a different email was used than the one they submitted
+    if (submittedEmail) {
+      const {contact_id} = await requestSignInEmail(submittedEmail)
+
+      cioIdentify(contact_id, {
+        email: submittedEmail,
+        [INTERESTED_IN_LIFETIME_SINCE]: Math.floor(Date.now() / 1000),
+      })
```

Comment:
if a user isn't signed in, they can enter their email which will send a sign in request to rails (effectively creating an account if there isn't one) which we can use the `contact_id` to identify the user in CIO